### PR TITLE
Power of 4

### PR DESCRIPTION
**Pull request by siddwarr**
### Describe your change:
Added the code for the leetcode problem 342
This is the most optimal solution using bit manipulation to check if a number is divisible by 4 or not

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: bit_manipulation/power_of_4.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    >>> power_of_4(17)
+    False
+    >>> power_of_4(-1)
+    Traceback (most recent call last):
+        ...
+    ValueError: number must be positive
+    >>> power_of_4(1.2)
+    Traceback (most recent call last):
+        ...
+    TypeError: unsupported operand type(s) for &: 'float' and 'float'
```
[the remark this answers is not in the document]
Comment (siddwarr) [reply bit_manipulation/power_of_4.py:45]: Added the type check

File: bit_manipulation/power_of_4.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    >>> power_of_4(2)
+    False
+    >>> power_of_4(4)
+    True
+    >>> power_of_4(6)
+    False
+    >>> power_of_4(8)
+    False
+    >>> power_of_4(17)
+    False
```
Comment (ChrisO345) [inline bit_manipulation/power_of_4.py:37]:
```suggestion
    False
    >>> power_of_4(64)
    True
```

**Review by ChrisO345 — CHANGES_REQUESTED** [review]

File: bit_manipulation/power_of_4.py (hunk 2 above)
Comment (siddwarr) [reply bit_manipulation/power_of_4.py:37]: added

**Comment by siddwarr** [comment]
added power_of_4(64) testcase

**Review by ChrisO345 — APPROVED** [review]

**Pull request merged**